src/main/scala/com/datastax/gatling/plugin/utils/CqlPreparedStatementUtil.scala

```diff
@@ -143,83 +168,106 @@ object CqlPreparedStatementUtil extends CqlPreparedStatementUtil {
+              }
+              case _ => throw new IllegalStateException("Observed something other than SetType for a SET param")
+            }
           case MAP =>
-            boundStatement.setMap(paramName, asMap(gatlingSession, paramName))
+            val dataType = bindable.getType(paramName)
+            dataType match {
+              case m: MapType => {
+                val keyClz = bindable.codecRegistry().codecFor(m.getKeyType).getJavaType.getRawType.asInstanceOf[Class[Any]]
+                val valClz = bindable.codecRegistry().codecFor(m.getValueType).getJavaType.getRawType.asInstanceOf[Class[Any]]
+                bindable.setMap(paramName, asMap(gatlingSession, paramName, keyClz, valClz), keyClz, valClz)
+              }
+              case _ => throw new IllegalStateException("Observed something other than MapType for a MAP param")
+            }
           case UDT =>
-            boundStatement.setUDTValue(paramName, asUdt(gatlingSession, paramName))
+            bindable.setUdtValue(paramName, asUdt(gatlingSession, paramName))
           case TUPLE =>
-            boundStatement.setTupleValue(paramName, asTuple(gatlingSession, paramName))
+            bindable.setTupleValue(paramName, asTuple(gatlingSession, paramName))
           case DATE =>
-            boundStatement.setDate(paramName, asDate(gatlingSession, paramName))
+            bindable.setLocalDate(paramName, asLocalDate(gatlingSession, paramName))
           case SMALLINT =>
-            boundStatement.setShort(paramName, asSmallInt(gatlingSession, paramName))
+            bindable.setShort(paramName, asSmallInt(gatlingSession, paramName))
           case TINYINT =>
-            boundStatement.setByte(paramName, asTinyInt(gatlingSession, paramName))
+            bindable.setByte(paramName, asTinyInt(gatlingSession, paramName))
           case TIME =>
-            boundStatement.setTime(paramName, asTime(gatlingSession, paramName))
+            bindable.setLocalTime(paramName, asTime(gatlingSession, paramName))
           case CUSTOM =>
             gatlingSession.attributes.get(paramName) match {
               case Some(p: Point) =>
-                boundStatement.set(paramName, asPoint(gatlingSession, paramName), classOf[Point])
+                bindable.set(paramName, asPoint(gatlingSession, paramName), classOf[Point])
               case Some(p: LineString) =>
-                boundStatement.set(paramName, asLineString(gatlingSession, paramName), classOf[LineString])
+                bindable.set(paramName, asLineString(gatlingSession, paramName), classOf[LineString])
               case Some(p: Polygon) =>
-                boundStatement.set(paramName, asPolygon(gatlingSession, paramName), classOf[Polygon])
+                bindable.set(paramName, asPolygon(gatlingSession, paramName), classOf[Polygon])
               case _ =>
                 throw new UnsupportedOperationException(s"$paramName on unknown CUSTOM type")
```

Comment:
The user might have registered a custom codec for this non-standard type. I think we should try to look up the codec and manually serialize here (as in the example I included in one of my previous comments).

If the custom type is truly unknown, the codec registry will throw a `CodecNotFoundException`.

Reply:
The SettableBy* interfaces support passing an explicit codec so can't we change this to the following and accomplish the same thing?

```scala
          case CUSTOM =>
            gatlingSession.attributes.get(paramName) match {
              case Some(p: Point) =>
                bindable.set(key, asPoint(gatlingSession, paramName), bindable.codecRegistry().codecFor(classOf[Point]))
              case Some(p: LineString) =>
                bindable.set(key, asLineString(gatlingSession, paramName), bindable.codecRegistry().codecFor(classOf[LineString]))
              case Some(p: Polygon) =>
                bindable.set(key, asPolygon(gatlingSession, paramName), bindable.codecRegistry().codecFor(classOf[Polygon]))
              case _ =>
                throw new UnsupportedOperationException(s"$paramName on unknown CUSTOM type")
            }
```

Reply:
I wasn't thinking in terms of new codecs for existing custom types, but rather new custom types. E.g. someone has their own `PriceType` on the server, and a `PriceCodec` added to their session.

For the existing geo types, the code above is ok.

Reply:
Ah, I see what you mean now.  Sounds like another useful enhancement for future.  Might be just a _little_ tricky since the match is largely used as a destructuring operation here which might make the types a bit complicated (if the user does something outside of the normal hierarchy) but that hardly seems intractable.